src/app/applications/applications.controller.new.js

```diff
@@ -0,0 +1,86 @@
+(function() {
+    'use strict';
+
+    // Declare the controllers first
+    angular
+        .module( 'roboconf.applications' )
+        .controller( 'ApplicationsNewController', applicationsNewController );
+
+    // Declare the controller functions then
+    // Specify the injection to prevent errors on minification
+    applicationsNewController.$inject = [ 'Restangular', '$scope', 'rAppTemplates', '$timeout', 'rShare' ];
+    
+    // Then comes the function
+    function applicationsNewController( Restangular, $scope, rAppTemplates, $timeout, rShare ) {
+    	
+    	// Fields
+    	$scope.appTemplates = [];
+    	$scope.fromExisting = true;
+    	$scope.app = {};
+    	$scope.errorMessage = '';
+    	
+    	// Functions declaration
+    	$scope.showFromExisting = showFromExisting;
+    	$scope.showUpload = showUpload;
+    	$scope.createNewApplication = createNewApplication;
+    	$scope.formatTpl = formatTpl;
+    	$scope.completeCreation = completeCreation;
+    	
+    	// Initialize the list of templates
+    	rAppTemplates.refreshTemplates().then( function() {
+    		$scope.appTemplates = rAppTemplates.getTemplates();
+    	});
+    	
+    	// Functions
+    	function showFromExisting() {
+        	$scope.fromExisting = true;
+        	$( '#upload-result-details' ).css( 'display', 'none' );
+        }
+        
+        function showUpload() {
+        	$scope.fromExisting = false;
+        }
+        
+        function createNewApplication( app ) {
+        	var newApp = {
+        		name: app.name,
+        		desc: app.description,
+        		tpl: {
+        			name: app.tpl.name,
+        			qualifier: app.tpl.qualifier
+        		}
+        	};
+        	
+        	Restangular.one( 'applications' ).post( '', newApp ).then( function() {
+        		rShare.feedLastItem( newApp );
+        		window.location = '#/';
```

Comment:
I am not sure to have understood the entire sentence.
But we use *window.location* to redirect to another page. We don't use $route or any Angular thing.

And this controller can be used for several URLs: **/**, **applications** and **application-templates**. The first two URL suffixes point to the same page. But the application itself only points to **/** in general. The second one is only there if a user typed it in manually.